src/std_extensions/arithmetic/conversions.rs

```diff
@@ -1,36 +1,40 @@
 //! Conversions between integer and floating-point values.
 
 use crate::{
-    extension::{ExtensionId, ExtensionSet, SignatureError},
+    extension::{ExtensionId, ExtensionRegistry, ExtensionSet, SignatureError, PRELUDE},
     type_row,
-    types::{type_param::TypeArg, FunctionType, Type},
-    utils::collect_array,
+    types::{FunctionType, PolyFuncType, Type},
     Extension,
 };
 
-use super::int_types::int_type;
+use super::int_types::{int_type_var, INT_TYPE_ID};
 use super::{float_types::FLOAT64_TYPE, int_types::LOG_WIDTH_TYPE_PARAM};
 
 /// The extension identifier.
 pub const EXTENSION_ID: ExtensionId = ExtensionId::new_unchecked("arithmetic.conversions");
 
-fn ftoi_sig(arg_values: &[TypeArg]) -> Result<FunctionType, SignatureError> {
-    let [arg] = collect_array(arg_values);
-    Ok(FunctionType::new(
+fn ftoi_sig(
+    int_type_var: Type,
+    temp_reg: &ExtensionRegistry,
+) -> Result<PolyFuncType, SignatureError> {
+    let body = FunctionType::new(
         type_row![FLOAT64_TYPE],
         vec![Type::new_sum(vec![
-            int_type(arg.clone()),
+            int_type_var,
```

Comment:
Consider a helper function `sum_with_error(Type) -> Type` (in some ancestor module I suspect)